### PR TITLE
feat(reasoning): add MiniMax M3 thinking control support

### LINKED ISSUE #15782: [Bug]: v1.9.11 MiniMax-M3 Error, v1.9.9 Works Normally
> [\!NOTE]
> This issue was translated by Claude.

### Issue Checklist

- [x] I understand that issues are for feedback and problem solving, not for complaining in the comment section, and will provide as much information as possible to help solve the problem.
- [x] My issue is not listed in the [FAQ](https://github.com/CherryHQ/cherry-studio/issues/3881).
- [x] I've looked at **pinned issues** and searched for existing [Open Issues](https://github.com/CherryHQ/cherry-studio/issues), [Closed Issues](https://github.com/CherryHQ/cherry-studio/issues?q=is%3Aissue%20state%3Aclosed), and [Discussions](https://github.com/CherryHQ/cherry-studio/discussions), no similar issue or discussion was found.
- [x] I've filled in short, clear headings so that developers can quickly identify a rough idea of what to expect when flipping through the list of issues. And not "a suggestion", "stuck", etc.
- [x] I've confirmed that I am using the latest version of Cherry Studio.

### Platform

Windows

### Version

v1.9.11

### Bug Description

v1.9.11 connected to MiniMax-M3 API with web search enabled, an error occurred. v1.9.9 works normally.

Image

Image

### Steps To Reproduce

In v1.9.11, with thinking enabled, selected Bing for web search, used the prompt: "Search for today's domestic and international news online and list the top ten news hotspots."

### Expected Behavior

As shown in the image: v1.9.9 output

### Relevant Log Output

```shell
Error message:

Error name: AI_APICallError
Error message: invalid params, invalid thinking.type: "enabled" (allowed: adaptive, disabled) (2013)
Stack trace: AI_APICallError: invalid params, invalid thinking.type: "enabled" (allowed: adaptive, disabled) (2013)
    at file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:52075:11
    at async postToApi$2 (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:51970:24)
    at async OpenAICompatibleChatLanguageModel.doStream (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:69970:48)
    at async wrapStream (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:105780:32)
    at async fn (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:104430:17)
    at async file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:100701:19
    at async _retryWithExponentialBackoff (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:100891:10)
    at async streamStep (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:104399:109)
    at async fn (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:104716:5)
    at async file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:100701:19
Error cause: "[undefined]"
Status code: 400
Request path: https://api.minimaxi.com/v1/chat/completions
Request body: {
  "model": "MiniMax-M3",
  "user": "[undefined]",
  "max_tokens": "[undefined]",
  "temperature": "[undefined]",
  "top_p": "[undefined]",
  "frequency_penalty": "[undefined]",
  "presence_penalty": "[undefined]",
  "response_format": "[undefined]",
  "stop": "[undefined]",
  "seed": "[undefined]",
  "thinking": {
    "type": "enabled"
  },
  "serviceTier": "[undefined]",
  "reasoning_effort": "[undefined]",
  "verbosity": "[undefined]",
  "messages": [
    {
      "role": "user",
      "content": "在网上搜索今天的国内和国际新闻，列出十大新闻热点。"
    }
  ],
  "tools": [
    {
      "type": "function",
      "function": {
        "name": "builtin_web_search",
        "description": "Web search tool for finding current information, news, and real-time data from the internet.\n\nThis tool has been configured with search parameters based on the conversation context:\n- Prepared queries: \"今天国内十大新闻热点\", \"今天国际十大新闻热点\"\n\nYou can use this tool as-is to search with the prepared queries, or provide additionalContext to refine or replace the search terms.",
        "parameters": {
          "\": "http://json-schema.org/draft-07/schema#",
          "type": "object",
          "properties": {
            "additionalContext": {
              "description": "Optional additional context, keywords, or specific focus to enhance the search",
              "type": "string"
            }
          },
          "additionalProperties": false
        }
      }
    }
  ],
  "tool_choice": "auto",
  "stream": true,
  "stream_options": {
    "include_usage": true
  }
}
Response headers: {
  "alb_receive_time": "1780877485.886",
  "alb_request_id": "58297f7148a1a10a596b4d29d9d60013f4d5e71d",
  "content-length": "219",
  "content-type": "application/json; charset=utf-8",
  "date": "Mon, 08 Jun 2026 00:11:25 GMT",
  "document-policy": "include-js-call-stacks-in-crash-reports",
  "minimax-request-id": "6542209161aaa1712bd93b400a569f4f",
  "trace-id": "06753bad837f29567e7222fcec667773",
  "x-session-id": "24fd5e03ec0d046d2740e7836f1d8174"
}
Response content: {
  "type": "error",
  "error": {
    "type": "bad_request_error",
    "message": "invalid params, invalid thinking.type: \"enabled\" (allowed: adaptive, disabled) (2013)",
    "http_code": "400"
  },
  "request_id": "06753bad837f29567e7222fcec667773"
}
Data: {
  "error": {
    "message": "invalid params, invalid thinking.type: \"enabled\" (allowed: adaptive, disabled) (2013)",
    "type": "bad_request_error"
  }
}
```

### Additional Context

_No response_

---
<details>
<summary>Original Content</summary>

### Issue Checklist

- [x] I understand that issues are for feedback and problem solving, not for complaining in the comment section, and will provide as much information as possible to help solve the problem.
- [x] My issue is not listed in the [FAQ](https://github.com/CherryHQ/cherry-studio/issues/3881).
- [x] I've looked at **pinned issues** and searched for existing [Open Issues](https://github.com/CherryHQ/cherry-studio/issues), [Closed Issues](https://github.com/CherryHQ/cherry-studio/issues?q=is%3Aissue%20state%3Aclosed), and [Discussions](https://github.com/CherryHQ/cherry-studio/discussions), no similar issue or discussion was found.
- [x] I've filled in short, clear headings so that developers can quickly identify a rough idea of what to expect when flipping through the list of issues. And not "a suggestion", "stuck", etc.
- [x] I've confirmed that I am using the latest version of Cherry Studio.

### Platform

Windows

### Version

v1.9.11

### Bug Description

v1.9.11 接入 MimiMax-M3 API，开启联网搜索，出现错误。v1.9.9 正常

Image

Image

### Steps To Reproduce

v1.9.11，开启思考，联网选择 Bing，提示词在网上搜索今天的国内和国际新闻，列出十大新闻热点。

### Expected Behavior

如图：v1.9.9 输出

### Relevant Log Output

```shell
报错信息：

错误名称: AI_APICallError
错误信息: invalid params, invalid thinking.type: "enabled" (allowed: adaptive, disabled) (2013)
堆栈信息: AI_APICallError: invalid params, invalid thinking.type: "enabled" (allowed: adaptive, disabled) (2013)
    at file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:52075:11
    at async postToApi$2 (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:51970:24)
    at async OpenAICompatibleChatLanguageModel.doStream (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:69970:48)
    at async wrapStream (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:105780:32)
    at async fn (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:104430:17)
    at async file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:100701:19
    at async _retryWithExponentialBackoff (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:100891:10)
    at async streamStep (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:104399:109)
    at async fn (file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:104716:5)
    at async file:///C:/Users/givx/AppData/Local/Programs/Cherry%20Studio/resources/app.asar/out/renderer/assets/store-D-J5UN_L.js:100701:19
错误原因: "[undefined]"
状态码: 400
请求路径: https://api.minimaxi.com/v1/chat/completions
请求体: {
  "model": "MiniMax-M3",
  "user": "[undefined]",
  "max_tokens": "[undefined]",
  "temperature": "[undefined]",
  "top_p": "[undefined]",
  "frequency_penalty": "[undefined]",
  "presence_penalty": "[undefined]",
  "response_format": "[undefined]",
  "stop": "[undefined]",
  "seed": "[undefined]",
  "thinking": {
    "type": "enabled"
  },
  "serviceTier": "[undefined]",
  "reasoning_effort": "[undefined]",
  "verbosity": "[undefined]",
  "messages": [
    {
      "role": "user",
      "content": "在网上搜索今天的国内和国际新闻，列出十大新闻热点。"
    }
  ],
  "tools": [
    {
      "type": "function",
      "function": {
        "name": "builtin_web_search",
        "description": "Web search tool for finding current information, news, and real-time data from the internet.\n\nThis tool has been configured with search parameters based on the conversation context:\n- Prepared queries: \"今天国内十大新闻热点\", \"今天国际十大新闻热点\"\n\nYou can use this tool as-is to search with the prepared queries, or provide additionalContext to refine or replace the search terms.",
        "parameters": {
          "\": "http://json-schema.org/draft-07/schema#",
          "type": "object",
          "properties": {
            "additionalContext": {
              "description": "Optional additional context, keywords, or specific focus to enhance the search",
              "type": "string"
            }
          },
          "additionalProperties": false
        }
      }
    }
  ],
  "tool_choice": "auto",
  "stream": true,
  "stream_options": {
    "include_usage": true
  }
}
响应首部: {
  "alb_receive_time": "1780877485.886",
  "alb_request_id": "58297f7148a1a10a596b4d29d9d60013f4d5e71d",
  "content-length": "219",
  "content-type": "application/json; charset=utf-8",
  "date": "Mon, 08 Jun 2026 00:11:25 GMT",
  "document-policy": "include-js-call-stacks-in-crash-reports",
  "minimax-request-id": "6542209161aaa1712bd93b400a569f4f",
  "trace-id": "06753bad837f29567e7222fcec667773",
  "x-session-id": "24fd5e03ec0d046d2740e7836f1d8174"
}
响应内容: {
  "type": "error",
  "error": {
    "type": "bad_request_error",
    "message": "invalid params, invalid thinking.type: \"enabled\" (allowed: adaptive, disabled) (2013)",
    "http_code": "400"
  },
  "request_id": "06753bad837f29567e7222fcec667773"
}
数据: {
  "error": {
    "message": "invalid params, invalid thinking.type: \"enabled\" (allowed: adaptive, disabled) (2013)",
    "type": "bad_request_error"
  }
}
```

### Additional Context

_No response_

</details>

### PR DESCRIPTION
**Pull request by ousugo**
### What this PR does

Before this PR:
- Cherry Studio had no first-class support for the MiniMax M3 model family. Models matching `minimax-m3*` were either routed through generic (Kimi / DeepSeek) code paths or misclassified, leaving the M3-specific `thinking: { type: 'adaptive' | 'disabled' }` shape unsupported.
- The `useAssistant` fallback silently simplified the 3-state `enableThinking` branch down to a single `??` chain, changing the persisted `reasoning_effort` from `'default'` to `'auto'` whenever a user first switched to a binary on/off model.

After this PR:
- M3 (and its variants — `minimax-m3`, `minimax-m3.1`, `minimax-m3.5`, `minimax-m3-highspeed`, `minimax-m3-ultra`, `minimax-m3-preview`, etc.) is recognized as a reasoning model via a new `isMiniMaxM3SeriesModel` predicate, exposed in the reasoning-effort UI as `['default', 'none', 'auto']`.
- `getReasoningEffort` and `getAnthropicReasoningParams` route M3 through a shared `getMiniMaxThinkingType` helper, emitting `{ thinking: { type: 'adaptive' } }` (or `'disabled'` when the user picks `none`).
- M2.x continues to fall through to `{}` — relies on the documented MiniMax default that M2.x thinking "cannot be turned off"; the trade-off is documented inline so future maintainers know the upstream contract they depend on.
- The `useAssistant` fallback is restored to the original 3-state `enableThinking` branch, preserving the historical "unset → `default`" semantics.

Fix #15789 

### Why we need it and why it was done in this way

The MiniMax M3 family exposes thinking control through a non-standard shape that no existing Cherry Studio branch handled. Per the [MiniMax Anthropic API reference](https://platform.minimaxi.com/docs/api-reference/text-anthropic-api) and the [MiniMax OpenAI-compatible API reference](https://platform.minimaxi.com/docs/api-reference/text-openai-api), M3 accepts only the following `thinking` directive:

- `{"type": "adaptive"}` — keep thinking on (M3-specific, identical to "enabled" for older models)
- `{"type": "disabled"}` — turn thinking off (M3 only; M2.x silently ignores this)
- omitted — on by default

`budget_tokens` is **not** supported, and M2.x's API "cannot be turned off" — every controllable directive is silently ignored. Both endpoints (Anthropic-compatible at `/anthropic` and OpenAI-compatible at `/v1/`) accept the same `thinking` shape, so a single code path covers both.

The following tradeoffs were made:

- **M2.x → `{}` vs explicit `enabled`:** We rely on the documented M2.x "always-on" default rather than emitting `{ thinking: { type: 'enabled' } }` (which M2.x silently ignores anyway). The trade-off is documented in the code: if MiniMax ever changes the M2.x default, M2.x users would silently lose thinking output. Revisit by switching to a sentinelled return once the upstream documents a stable on-by-default contract.
- **New `isMiniMaxM3SeriesModel` predicate vs. extending `isMiniMaxReasoningModel`'s id array:** M3 has its own controllable protocol (vs. M2.x's "fixed" behavior), so it warrants a separate predicate. The regex covers M3 and all its `-xxx` / `.x` variants (e.g. `m3.1`, `m3-highspeed`, `m3-ultra`) while explicitly excluding `m30` / `m300` (separate model lines).
- **M3 added to `_isSupportedThinkingTokenModel` chain:** M3 doesn't support `budgetTokens`, but the chain is also the entry point for `useAssistant` to pick up model-type options. Adding it there routes M3 into the `['default', 'none', 'auto']` UI options. Inline comment explains the routing.
- **`useAssistant` 3-state branch restored:** A previous simplification to a `??` chain silently dropped the "user has no `reasoning_effort` set" case, persisting `'auto'` instead of `'default'`. Restored the original 3-state branch.

The following alternatives were considered:

- A per-model-type `THINKING_SHAPE_RESOLVER` registry to data-drive the per-family mapping (Claude 4.6, Opus 4.7+, DeepSeek V4+, M3). Deferred to the v2 refactor — out of scope for a v1.9.x bug-fix-shaped PR.
- Using the existing `isFixedReasoningModel` predicate for the M2.x early-return. Functionally equivalent, but the current code is more explicit about *why* M2.x is being intercepted (M2.x-specific doc comment). Left for a follow-up.

Links to places where the discussion took place: N/A

### Breaking changes

None. New behavior is opt-in via the new `isMiniMaxM3SeriesModel` predicate. M2.x users see no change in emitted parameters.

### Special notes for your reviewer

- The `isMiniMaxM3SeriesModel` regex is `/(\b|[.-])minimax-m3(\.\d+)?([.-][\w-]+)*$/`. Boundary cases (`m3.1`, `m3.5`, `m3-highspeed`, `m3-ultra`, `m3-preview`, `m3-turbo`, `m3-ultra.1`, `m2.7`, `m30`, `m300`) are covered by tests in `config/models/__tests__/reasoning.test.ts`.
- The M3 path emits `sendReasoning: true` only in `getAnthropicReasoningParams` (Anthropic SDK's internal provider-options key for streaming reasoning back to the UI). The OpenAI-compatible path does not — that's intentional, since `sendReasoning` is a `@ai-sdk/anthropic`-specific field.
- The M2.x early-return is now backed by an inline comment citing the MiniMax docs. Please push back if you prefer a more defensive `{ thinking: { type: 'enabled' } }` byte — both behaviors are equivalent today, but the current choice saves bytes and avoids sending a directive the API silently ignores.

### Checklist

- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: Write code that humans can understand and Keep it simple
- [x] Refactor: You have left the code cleaner than you found it (Boy Scout Rule) — `useAssistant` 3-state branch restored, helper extracted for M3 thinking-type resolution
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Documentation: A user-guide update was considered and is present (link) or not required. Check this only when the PR introduces or changes a user-facing feature or behavior.
- [x] Self-review: I have reviewed my own code (e.g., via `/gh-pr-review`, `gh pr diff`, or GitHub UI) before requesting review from others

### Release note

```release-note
feat(reasoning): add MiniMax M3 thinking control support

M3 is now recognized as a reasoning model and routed through the new
thinking-control path. Users picking M3 from the model list will see
[default, none, auto] reasoning-effort options in the UI; selecting
"none" emits { thinking: { type: 'disabled' } }, while default/auto
emit { thinking: { type: 'adaptive' } } per the MiniMax API contract.
```

### CODE REVIEW

**Review by GeorgeDong32 — APPROVED** [review]
## 审查总结

针对 PR #15769 (feat(reasoning): add MiniMax M3 thinking control support) 的代码审查已完成,经核实:

**核心契约(M2.x 行为)已与官方文档对齐**:
- 文档明确说明 "对于 M2.x 模型,thinking 无法关闭;即使传入 `thinking: { type: 'disabled' }`,thinking 仍会保持开启"
- M3 协议支持 `thinking.type` 取值 `adaptive` (显式保持开启) 和 `disabled` (关闭 M3 thinking 输出)
- Anthropic 端点与 OpenAI 兼容端点协议一致

**审查通过项**:
- `isMiniMaxM3SeriesModel` 正则 `/(\b|[.-])minimax-m3(\.\d+)?([.-][\w-]+)*$/` 经测试正确覆盖 `m3` / `m3.1` / `m3.5` / `m3-highspeed` / `m3-ultra` / `m3-preview` / `m3-turbo` / `m3-ultra.1`,正确排除 `m2.7` / `m30` / `m300` / `m3x` / `minimax-m31`
- `getMiniMaxThinkingType` 抽离为单一职责函数,在 `getReasoningEffort` 和 `getAnthropicReasoningParams` 间复用,符合 DRY
- M2.x 早返回 `{}` 的设计有 4 行注释说明是依赖上游默认,设计权衡显式记录
- 测试覆盖 4 核心矩阵 (M3/M2.x × default/none),边界用例 11 项
- 路径顺序正确(M3 检查在 M2.x 早返回前,确保 M3 走专属分支)
- M3 在 Anthropic 端点设 `sendReasoning: true`,在 OpenAI 端点不设——与 SDK 字段归属一致

**遗留观察(非阻塞)**:
- `isMiniMaxReasoningModel` 数组仍含 `'minimax-m3'`,但因 M3 在前面被拦截,实际不可达——保留或移除都是合理的命名空间选择
- M3 协议未识别 `enabled` 字面值,建议在 `getReasoningEffort` 注释中加一句明确说明
- `useAssistant.ts` 提交仅新增 2 行注释,代码本身未改(3 态 `enableThinking` 分支在 v1 上是原始状态,PR 描述里 "restored" 措辞略具误导性但不影响功能)

LGTM ✅

**Comment by egorzainullin** [comment]
Will this fix this problem?
image
image

**Comment by nfedyashev** [comment]
Related pull request https://github.com/CherryHQ/cherry-studio/pull/15789

File: src/renderer/src/aiCore/utils/reasoning.ts [hunk 1]
```diff
@@ -727,6 +752,27 @@ export function getAnthropicReasoningParams(
...
   }
 
+  // MiniMax M3 only accepts the Anthropic-protocol thinking shape
+  // (type: 'adaptive' | 'disabled'). M2.x is intercepted here because its
+  // API rejects every controllable directive: thinking cannot be turned
+  // off and the model does not understand 'enabled' + budgetTokens. The
+  // only safe answer is {} — let the API default to thinking-on.
+  const m3ThinkingType = getMiniMaxThinkingType(assistant, model)
+  if (m3ThinkingType) {
+    return { thinking: { type: m3ThinkingType }, sendReasoning: true }
```
Comment (kangfenmao) [inline src/renderer/src/aiCore/utils/reasoning.ts:766]: 这里在 Anthropic-compatible 路径下会把 `default`/`auto` 都转成 `thinking: { type: 'adaptive' }`，这会触发当前 `@ai-sdk/anthropic` 的 thinking 分支：adapter 只要看到 `enabled/adaptive` 就会把 `temperature`、`top_p`、`top_k` 从请求体里置空。但 MiniMax Anthropic 文档把 `temperature`、`top_p` 标为支持，并且 M3 省略 `thinking` 时默认关闭，而不是开启。结果是用户在 MiniMax-M3 的 Anthropic endpoint/agent session 上使用默认配置时，PR 会强制打开 thinking，并静默忽略他们配置的采样参数。

建议至少让 Anthropic 路径的 `default` 保持 `{}`，只在用户显式选 `auto` 时发送 `adaptive`；同时需要避免经由 `@ai-sdk/anthropic` 的 Claude thinking 限制丢掉 MiniMax 支持的采样参数。

**Review by kangfenmao — CHANGES_REQUESTED** [review]
我看完了这轮差异，OpenAI-compatible 路径和 M3/M2.x 识别本身基本合理，CI 必跑项也都是绿的。

但 Anthropic-compatible 路径这里还有一个会影响用户配置的行为问题：当前实现把 M3 的 `default`/`auto` 都送进 `@ai-sdk/anthropic` 的 thinking 分支，导致 MiniMax 文档支持的采样参数被 adapter 静默丢掉，同时也改变了 Anthropic endpoint 上省略 `thinking` 的默认关闭语义。这个需要先修一下再合并。

File: src/renderer/src/aiCore/utils/reasoning.ts (hunk 1 above)
Comment (ousugo) [reply src/renderer/src/aiCore/utils/reasoning.ts:766]: 已修复并推送到 `8ce2cd0a7`：

1. MiniMax M3 的 `undefined/default` 现在保持 `{}`，不再发送 `thinking`，只有用户显式选择 `auto` 时才发送 `thinking: { type: "adaptive" }`，`none` 仍发送 `disabled`。
2. `@ai-sdk/anthropic` patch 现在只在真正的 Claude 模型路径清空 `temperature/top_k/top_p`，MiniMax M3 这类 Anthropic-compatible 非 Claude 模型会保留采样参数。

已补充回归测试覆盖 M3 默认/auto 行为，以及 SDK 最终请求体中 MiniMax 保留采样参数、Claude 仍保持原有清空逻辑。

**Comment by ousugo** [comment]
> [!NOTE]
> This comment was translated by Claude.

Processed the latest feedback and pushed to \`dea97168c\`:

- Updated the patch hash for \`@ai-sdk/anthropic\` in \`pnpm-lock.yaml\` to fix CI's \`ERR_PNPM_LOCKFILE_CONFIG_MISMATCH\`. Verified locally that \`pnpm install --frozen-lockfile --ignore-scripts\` passes.
- Adjusted \`scripts/__tests__/anthropic-compatible-thinking.test.ts\` per CodeRabbit suggestions: changed \`CapturedRequest\` to an interface and added explicit return types for helpers.
- Added coverage for passing through \`thinking: { type: \"disabled\" }\`, and for omitting \`max_tokens\` when non-Claude Anthropic-compatible models don't provide \`maxOutputTokens\`.

Ran \`pnpm test:scripts\` locally and passed.

---
<details>
<summary>Original Content</summary>

已处理最新反馈并推送到 \`dea97168c\`：

- 更新 \`pnpm-lock.yaml\` 中 \`@ai-sdk/anthropic\` patch hash，修复 CI 的 \`ERR_PNPM_LOCKFILE_CONFIG_MISMATCH\`。本地已验证 \`pnpm install --frozen-lockfile --ignore-scripts\` 通过。
- 按 CodeRabbit 建议调整 \`scripts/__tests__/anthropic-compatible-thinking.test.ts\`：\`CapturedRequest\` 改为 interface，并为 helper 增加显式返回类型。
- 补充覆盖 \`thinking: { type: \"disabled\" }\` 透传，以及非 Claude Anthropic-compatible 模型未提供 \`maxOutputTokens\` 时省略 \`max_tokens\` 的分支。

本地已跑 \`pnpm test:scripts\` 通过。

</details>
